src/coreclr/jit/lsrabuild.cpp

```diff
@@ -2125,6 +2129,8 @@ void LinearScan::buildIntervals()
     }
 #endif // DEBUG
 
+    resetRegState();
```

Comment:
What's the reason for needing to `resetRegState()` when doing `buildIntervals` that we didn't need before?